### PR TITLE
Fix and test filter stream_tags.

### PR DESCRIPTION
**Pull request by postwait**
* metric matching only matches against the metric base name (not tags).
 * __name is inserted and made available to the stream_tags expression.

### CODE REVIEW

File: src/noit_filters.c [hunk 1]
```diff
@@ -482,7 +495,7 @@ noit_apply_filterset(const char *filterset,
       need_target = !MATCHES(target, check->target);
       need_module = !MATCHES(module, check->module);
       need_name = !MATCHES(name, check->name);
-      need_metric = !noit_apply_filterrule_metric(r, metric->metric_name, strlen(metric->metric_name), &stset, &mtset);
+      need_metric = !noit_apply_filterrule_metric(r, metric->metric_name, mlen, &stset, &mtset);
```
Comment (ewilhelm) [inline src/noit_filters.c]: IIUC, this is taking the mlen from noit_metric_parse_tags(), which is the length of the metric name without tags?

Comment (bdunavant) [reply src/noit_filters.c]: noit_metric_parse_tags() returns the input len passed into it, which is `strlen(metric->metric_name)`.  It's the same value, we just aren't recomputing it a second time.

Comment (ewilhelm) [reply src/noit_filters.c]: ...and noit_apply_filterrule_metric() already supported stream tags `stsearch`

File: src/noit_check_rest.c [hunk 2]
```diff
@@ -101,6 +101,9 @@ add_metrics_to_node(stats_t *c, xmlNodePtr metrics, const char *type,
     xmlSetProp(tmp, (xmlChar *)"name", (xmlChar *)m->metric_name);
     buff[0] = m->metric_type; buff[1] = '\0';
     xmlSetProp(tmp, (xmlChar *)"type", (xmlChar *)buff);
+    if(!noit_apply_filterset(check->filterset, check, m)) {
+      xmlSetProp(tmp, (xmlChar *)"filtered", (xmlChar *)"true");
+    }
```
Comment (ewilhelm) [inline src/noit_check_rest.c]: I don't see any comment about this new filtered / _filtered attribute in the xml or json

File: src/noit_filters.c (hunk 1 above)
Comment (postwait) [reply src/noit_filters.c]: noit_metric_parse_tags returns the length without andy tags.  The "eat_up_tags" function in there modifies (down) input_len as it ... eats up tags.

Comment (bdunavant) [reply src/noit_filters.c]: Ah, i missed the `eat_up_tags(input, &input_len`

**Comment by postwait** [comment]
addressed

File: src/noit_check_rest.c (hunk 2 above)
Comment (postwait) [reply src/noit_check_rest.c]: I noticed that the check pulling has no way to show which metrics are currently filtered out.  so I added an attribute.  I didn't add the "false" case b/c the documents are large and didn't need to get much larger.

**Pull request merged**